src/components/LoginBox/LoginBox.js

```diff
@@ -4,34 +4,49 @@ import { Row } from 'react-bootstrap';
 import { withRouter, NavLink } from 'react-router-dom';
 import { loginUser } from '../../redux/actions';
 import { connect } from 'react-redux';
-
 import './LoginBox.css';
 
 class LoginBox extends PureComponent {
-
   state = {
     email: '',
     password: '',
     errors: {}
-  }
+  };
 
   handleLogin = e => {
+    const {
+      loginUser,
+      history,
+    } = this.props;
+
+    const {
+      email,
+      password,
+    } = this.state;
```

Comment:
Destructuring: use it! If you ever see yourself using `anyObject.` more than once, destructure.